cmd/publishing-bot/config/rules_example.yaml

```diff
@@ -0,0 +1,686 @@
+rules:
+- destination-repository: apimachinery
+  source-to-destination:
+  - source:
+      repository: kubernetes
+      branch: master
+      dir: staging/src/k8s.io/apimachinery
+    destination:
+      repository: apimachinery
```

Comment:
isn't this the same as the `destination-repository` above?

Reply:
Would such a struct do?
```yaml
rules:
- destination: apimachinery
   branches:
     master:
       source: kubernetes
       source-branch: master
       dir: staging/src/k8s.io/apimachinery
       destination-dir: ./  # optional
```

Reply:
branches can't be map as they have to be ordered, but I tried to simplify this a little.

Reply:
why do they have to be ordered?